django_adobesign/client.py

```diff
@@ -54,6 +53,17 @@ def refresh_token(self, refresh_token, application_secret):
         return response
 
 
+def handle_adobe_exception(function):
+    def wrapper(*arg, **kwargs):
+
+        try:
+            return function(*arg, **kwargs)
+        except (requests.exceptions.RequestException, HTTPError) as e:
+            raise AdobeExceptionFactory.get_exception(e)
+
+    return wrapper
```

Comment:
Hm :/ You should use [functools.wraps](https://docs.python.org/3.7/library/functools.html#functools.wraps) whenever creating a decorator and that makes the code quite less readable.

An alternative would be using a contextmanager (through [contexlib.contextmanager](https://docs.python.org/3.7/library/contextlib.html#contextlib.contextmanager)) but that would change the resulting code.

Reply:
I think the decorator is the most appropriate in this case.
About the functools.wraps I'm not sure I understood your argument but I think it is about docstring and method name. 
Indeed without it we have:
`
>>> from django_adobesign.client import AdobeSignClient
>>> help(AdobeSignClient)
Help on class AdobeSignClient in module django_adobesign.client:
class AdobeSignClient(builtins.object)
 |  AdobeSign client use v6 api.
 |  See https://secure.na1.echosign.com/public/docs/restapi/v6
 |  
 |  Methods defined here:
 |  
 |  __init__(self, root_url, access_token, api_user=None, on_behalf_of_user=None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |  
 |  build_url(self, urlpath)
 |  
 |  get_agreements = wrapper(*arg, **kwargs)
 |  
 |  get_document = wrapper(*arg, **kwargs)
 |
[...] 
And using it: 
class AdobeSignClient(builtins.object)
 |  AdobeSign client use v6 api.
 |  See https://secure.na1.echosign.com/public/docs/restapi/v6
 |  
 |  Methods defined here:
 |  
 |  __init__(self, root_url, access_token, api_user=None, on_behalf_of_user=None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |  
 |  build_url(self, urlpath)
 |  
 |  get_agreements(self, page_size, cursor=None, **extra_params)
 |      Return the list of agreements with pagination
 |  
 |  get_document(self, agreement_id, document_id)
 |      Download a document
`

See [also](https://www.blog.pythonlibrary.org/2016/02/17/python-functools-wraps/)

Reply:
+1 for contextmanager if possible :)